app/code/core/Mage/Adminhtml/Block/Sales/Items/Abstract.php

```diff
@@ -562,10 +562,7 @@ public function canShipPartially($order = null)
             $order = Mage::registry('current_shipment')->getOrder();
         }
         $value = $order->getCanShipPartially();
-        if (!is_null($value) && !$value) {
-            return false;
-        }
-        return true;
+        return !(!is_null($value) && !$value);
```

Comment:
DeMorgan's Law `!(A && B) == (!A || !B)`:

```suggestion
        return is_null($value) || $value;
```